### PR TITLE
User Authentication

### PR DESCRIPTION
**Pull request by mnath30**
Added the following:
1. Login page
2. Signup page
3. Logout page
4. Api calls for login, signup 
5. Loader component

### CODE REVIEW

File: src/pages/Authentication/Signup.jsx [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+    <div className="auth-main">
+      {loading && <Loader />}
+      {!loading && (
+        <div className="auth-container">
+          <h3 className="txt-center txt-lg form-heading">Signup</h3>
+          <form className="txt-form">
+            <label htmlFor="name" className="txt-md form-label">
+              Name
+            </label>
+            <input
```
Comment (vashnavichauhan1825) [inline src/pages/Authentication/Signup.jsx:60]: suggestion :- use implicit label (wrap input around label)

**Review by vashnavichauhan1825 — COMMENTED** [review]
code is well structured and readable

**Review by anjali1102 — APPROVED** [review]
working good.
ready to merge

File: src/pages/Authentication/Signup.jsx (hunk 1 above)
Comment (mnath30) [reply src/pages/Authentication/Signup.jsx:60]: Thankyou for your suggestion will implement this in next pr

**Pull request merged**